src/Stryker.Core/Stryker.Core/MutantFilters/DiffMutantFilter.cs

```diff
@@ -78,23 +78,33 @@ public IEnumerable<Mutant> FilterMutants(IEnumerable<Mutant> mutants, FileLeaf f
                 UpdateMutantsWithBaselineStatus(mutants, file);
             }
 
-            // A non-cSharp file is flagged by the diff result as modified. We cannot determine which mutants will be affected by this, thus all mutants have to be tested.
-            if (_diffResult.TestFilesChanged is { } && _diffResult.TestFilesChanged.Any(x => !x.EndsWith(".cs")))
+
+
+            // A non-cSharp file is flagged by the diff result as modified. We cannot determine which mutants will be affected by this, thus all mutants have to be tested, except when the changed files are excluded.
+            if (_diffResult.TestFilesChanged is { })
```

Comment:
I don't think this option should be exclusive to the diff mutant filter, it might be better to do the filtering in diffProvider.ScanDiff() and simply not return any files that match the pattern (test AND source files).